.circleci/config.yml

```diff
@@ -36,181 +36,175 @@ defaults_license_scanner: &defaults_license_scanner
 
 defaults_npm_auth: &defaults_npm_auth
   name: Update NPM registry auth token
-  command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > src/.npmrc
+  command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc
 
 defaults_npm_publish_version: &defaults_npm_publish
   name: Update version to prerelease
   command: |
     source $BASH_ENV
     echo "Publishing tag $CIRCLE_TAG"
-    cd src && npm publish --tag $CIRCLE_TAG --access public
+    npm publish --tag $CIRCLE_TAG --access public
 
 defaults_npm_publish_release: &defaults_npm_publish_release
   name: Publish NPM $RELEASE_TAG artifact
   command: |
     source $BASH_ENV
     echo "Publishing tag $RELEASE_TAG"
-    cd src && npm publish --tag $RELEASE_TAG --access public
+    npm publish --tag $RELEASE_TAG --access public
+
+defaults_export_version_from_package: &defaults_export_version_from_package
+  # TODO: this is rather imperfect, but will do for now
+  name: Format the changelog into the github release body and get release tag
+  command: |
+    git diff --no-indent-heuristic master~1 HEAD CHANGELOG.md | sed -n '/^+[^+]/ s/^+//p' > /tmp/changes
+    echo 'export RELEASE_CHANGES=`cat /tmp/changes`' >> $BASH_ENV
+    echo 'export RELEASE_TAG=`cat package-lock.json | jq -r .version`' >> $BASH_ENV
 
-defaults_slack_announcement: &defaults_slack_announcement
-  name: Slack announcement for tag releases
+defaults_configure_git: &defaults_configure_git
+  name: Configure git
   command: |
-    curl -X POST \
-      $SLACK_WEBHOOK_ANNOUNCEMENT \
-      -H 'Content-type: application/json' \
-      -H 'cache-control: no-cache' \
-      -d "{
-      \"text\": \"*${CIRCLE_PROJECT_REPONAME}* - Release \`${CIRCLE_TAG}\`: https://github.com/mojaloop/${CIRCLE_PROJECT_REPONAME}/releases/tag/${CIRCLE_TAG}\"
-    }"
+    git config user.email ${GIT_CI_EMAIL}
+    git config user.name ${GIT_CI_USER}
+
+##
+# Executors
+#
+# CircleCI Executors
+##
+executors:
+  default-docker:
+    working_directory: &WORKING_DIR /home/circleci/project
+    docker:
+      - image: node:16.14.2-alpine
+
+  default-machine:
+    working_directory: *WORKING_DIR
+    machine:
+      image: ubuntu-2004:current
 
 jobs:
   setup:
-    <<: *defaults_working_directory
-    <<: *defaults_docker_node
+    executor: default-docker
     steps:
       - run:
           name: Install general dependencies
           command: *defaults_Dependencies
       - checkout
       - run:
           name: Update NPM install
-          command: cd src && npm ci
+          command: npm install
```

Comment:
Shouldn't we use `npm ci`? `npm install` doesn't respect the `package-lock.json`

Reply:
Good point